python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -172,6 +175,37 @@ def resolve_partitionKeys(self, _graphene_info):
                 ]
         return []
 
+    def resolve_latestMaterializationPerPartition(self, _graphene_info, **kwargs):
+        from ..implementation.fetch_assets import get_asset_events
+
+        get_timestamp = lambda event: event.timestamp
+        get_partition = (
+            lambda event: event.dagster_event.step_materialization_data.materialization.partition
+        )
+
+        partitions = kwargs.get("partitions")
+        events_for_partitions = get_asset_events(
```

Comment:
If we could get these returned in newest => oldest order, it'd make the for loop below a lot cleaner since the "else:" on 198 could just continue and the first data we see is the data we want. Or if they're sorted the other way we could `.reverse()` the set and then iterate?

Reply:
Edit - it looks like there is an ascending=False default on here, so maybe they're already sorted!